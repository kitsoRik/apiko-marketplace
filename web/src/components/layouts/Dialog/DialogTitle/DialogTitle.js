import React from "react";

import "./DialogTitle.scss";

const DialogTitle = ({ value, className, ...props }) => {
	return (
		<h2 className={`dialog-title ${className ?? ""}`} {...props}>
			{value}
		</h2>
	);
};

export default DialogTitle;
