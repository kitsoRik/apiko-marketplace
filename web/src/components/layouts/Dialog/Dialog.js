import React from 'react';

import "./Dialog.scss";
import Form from '../Form';
import DialogTitle from './DialogTitle';

const Dialog = ({ onClose, open = true, children, className, ...props }) => {

    const ch = children ? [children] : [];

    const title = ch.find(c => c.type === DialogTitle);

    const other = ch.filter(c => c.type !== DialogTitle);

    return (
        <div
            className={`dialog ${className ?? ""}`}
            onClick={onClose}
            style={{ visibility: open ? "visible" : "hidden" }}
            {...props} >
            <Form onClick={e => e.stopPropagation()}>
                {title}
                {other}
            </Form>
        </div>
    )
};

export default Dialog;