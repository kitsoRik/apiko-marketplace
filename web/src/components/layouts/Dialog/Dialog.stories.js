import React from 'react';
import Dialog from './Dialog';
import TextField from '../TextField';

export const Default  = () => <Dialog>
    <TextField />
</Dialog>

export default { title: "Dialog" };