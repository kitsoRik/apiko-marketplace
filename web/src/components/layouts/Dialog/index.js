import Dialog from "./Dialog";
import DialogTitle from "./DialogTitle";

export const Title = DialogTitle;

export default Dialog;
