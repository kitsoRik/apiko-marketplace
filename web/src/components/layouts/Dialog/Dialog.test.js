import React from "react";
import { create } from "react-test-renderer";
import Dialog from "./Dialog";
import DialogTitle from "./DialogTitle";

describe("Dialog", () => {
	it("should render without children", () => {
		const component = create(<Dialog />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with title", () => {
		const component = create(
			<Dialog>
				<DialogTitle />
			</Dialog>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with title and content", () => {
		const component = create(
			<Dialog>
				<DialogTitle />
				<div>Content</div>
			</Dialog>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
	it("should render with content", () => {
		const component = create(
			<Dialog>
				<div>Content</div>
			</Dialog>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
});
