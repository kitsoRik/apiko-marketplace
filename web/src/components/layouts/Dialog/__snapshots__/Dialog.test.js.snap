// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should render with content 1`] = `
<div
  className="dialog "
  style={
    Object {
      "visibility": "visible",
    }
  }
>
  <div
    className="form "
    onClick={[Function]}
  >
    <div>
      Content
    </div>
  </div>
</div>
`;

exports[`Dialog should render with title 1`] = `
<div
  className="dialog "
  style={
    Object {
      "visibility": "visible",
    }
  }
>
  <div
    className="form "
    onClick={[Function]}
  >
    <h2
      className="dialog-title "
    />
  </div>
</div>
`;

exports[`Dialog should render with title and content 1`] = `
<div
  className="dialog "
  style={
    Object {
      "visibility": "visible",
    }
  }
>
  <div
    className="form "
    onClick={[Function]}
  >
    <h2
      className="dialog-title "
    />
    <div>
      Content
    </div>
  </div>
</div>
`;

exports[`Dialog should render without children 1`] = `
<div
  className="dialog "
  style={
    Object {
      "visibility": "visible",
    }
  }
>
  <div
    className="form "
    onClick={[Function]}
  />
</div>
`;
